engines/osxnative/engine.go

```diff
@@ -49,6 +67,33 @@ var payloadSchema = schematypes.Object{
 	Required: []string{"command"},
 }
 
+var configSchema = schematypes.Object{
+	Properties: schematypes.Properties{
+		"createUser": schematypes.Boolean{
+			MetaData: schematypes.MetaData{
+				Title: "Tells if a user should be created to run a command",
+				Description: `When set to true, a new user is created on the fly to run
+				the command. It runs the command from whitin the user's home directory.`,
+			},
+		},
+		"userGroups": schematypes.Array{
+			MetaData: schematypes.MetaData{
+				Title: "List of groups to assign user to",
+				Description: `It contains the list of groups the user will belong to.
+				The first item is the user's primary group, the rest will be user's
+				supplementary groups.`,
+			},
+			Items:  schematypes.String{},
```

Comment:
I was just suggesting that you restrict it to a pattern like: `Pattern: "^[a-zA-Z0-9_][a-zA-Z0-9_-]*$"`

So that the group names can't start with dash, contain spaces or other "fun" things that could potentially cause issues if used as command line arguments... Anyways, I guess it's not really an attack vector as this is part of engine config, and anyone writing engine config has full system access.